README.md

```diff
@@ -484,6 +484,86 @@ client.Do(ctx, client.B().FtSearch().Index("idx").Query("@f:v").Build()).AsFtSea
 client.Do(ctx, client.B().Geosearch().Key("k").Fromlonlat(1, 1).Bybox(1).Height(1).Km().Build()).AsGeosearch()
 ```
 
+## Use DecodeSliceOfJSON to scan array result
+
+DecodeSliceOfJSON is useful when you would like to scan the results of an array into a slice of a specific struct.
+
+```golang
+type User struct {
+	Name string
+}
+
+// Set some values
+if err = client.Do(ctx, client.B().Set().Key("user1").Value(`{"Name": "name1"}`).Build()).Error(); err != nil {
+	return err
+}
+if err = client.Do(ctx, client.B().Set().Key("user2").Value(`{"Name": "name2"}`).Build()).Error(); err != nil {
+	return err
+}
+
+// Scan MGET results into []*User
+users := make([]*User, 0) // []User is also being scannable
+if err := rueidis.DecodeSliceOfJSON(client.Do(ctx, client.B().Mget().Key("user1", "user2").Build()), &users); err != nil {
+	return err
+}
+
+for _, user := range users {
+	fmt.Printf("%+v\n", user)
+}
+/*
+&{Name:name1}
+&{Name:name2}
+*/
+```
+
+Implement UnmarshalJSON to scan private fields
+
+```golang
+type User struct {
+	name string
+}
+
+func (u *User) UnmarshalJSON(b []byte) error {
+	tmp := struct {
+		Name string
+	}{}
+
+	if err := json.Unmarshal(b, &tmp); err != nil {
+		return err
+	}
+
+	u.name = tmp.Name
+	return nil
+}
+
+if err = client.Do(ctx, client.B().Set().Key("user1").Value(`{"name": "name1"}`).Build()).Error(); err != nil {
+	return err
+}
+
+
+users := make([]*User, 0)
+if err := rueidis.DecodeSliceOfJSON(client.Do(ctx, client.B().Mget().Key("user1").Build()), &users); err != nil {
+	return err
+}
+// users = [{name:name1}]
+```
```

Comment:
I wonder if this section is necessary...
for some experienced golang developer don't need it but some people like beginners or not familiar with golang might do.

Reply:
Indeed, I think this section is not necessary, and scanning into private fields is also not a common approach. Actually using `json` tags is preferred in the above case.